.azure-pipelines/ci-build.yml

```diff
@@ -0,0 +1,265 @@
+                - pwsh: dotnet pack "$(ProjectFile)" /p:IncludeSymbols=true /p:SymbolPackageFormat=snupkg --no-build --output $env:BUILD_ARTIFACTSTAGINGDIRECTORY/Nugets --configuration $env:BUILD_CONFIGURATION
+                  env:
+                      BUILD_CONFIGURATION: $(BuildConfiguration)
+                  displayName: Dotnet pack
+
+                - task: PowerShell@2
+                  displayName: "Validate project version has been incremented"
+                  condition: and(contains(variables['build.sourceBranch'], 'refs/heads/main'), succeeded())
+                  inputs:
+                      targetType: "filePath"
+                      filePath: $(System.DefaultWorkingDirectory)\scripts\ValidateProjectVersionUpdated.ps1
+                      arguments: '-projectPath "$(ProjectFile)" -packageName "Microsoft.Graph.Cli.Core"'
+                      pwsh: true
+
+                - task: SFP.build-tasks.custom-build-task-1.EsrpCodeSigning@1
+                  displayName: "ESRP CodeSigning Nuget Packages"
+                  inputs:
+                      ConnectedServiceName: "microsoftgraph ESRP CodeSign DLL and NuGet (AKV)"
+                      FolderPath: "$(Build.ArtifactStagingDirectory)/Nugets"
+                      Pattern: "*.nupkg"
+                      signConfigType: inlineSignParams
+                      inlineOperation: |
+                          [
+                              {
+                                  "keyCode": "CP-401405",
+                                  "operationSetCode": "NuGetSign",
+                                  "parameters": [ ],
+                                  "toolName": "sign",
+                                  "toolVersion": "1.0"
+                              },
+                              {
+                                  "keyCode": "CP-401405",
+                                  "operationSetCode": "NuGetVerify",
+                                  "parameters": [ ],
+                                  "toolName": "sign",
+                                  "toolVersion": "1.0"
+                              }
+                          ]
+                      SessionTimeout: 20
+
+                - task: CopyFiles@2
+                  displayName: "Copy release scripts to artifact staging directory"
+                  condition: and(contains(variables['build.sourceBranch'], 'refs/heads/main'), succeeded())
+                  inputs:
+                      SourceFolder: "$(Build.SourcesDirectory)"
+                      Contents: 'scripts\**'
+                      TargetFolder: "$(Build.ArtifactStagingDirectory)/Nugets"
+
+                - task: PublishPipelineArtifact@1
+                  displayName: "Upload Artifact: Nugets"
+                  inputs:
+                      artifactName: Nugets
+                      targetPath: $(Build.ArtifactStagingDirectory)/Nugets
+
+    - stage: deploy
```

Comment:
I think we should make this a stage to drop an artifact. I prefer that we not deploy from an Azure Pipeline and that we use an Azure Release. My reasons are:
* Decouple build+sign from the release. Separate the concerns. 
* Capture of release artifacts for review before deployment. Also provides us information about historical release artifacts in case we need to go back and investigate releases.
* Notification and signoff before release. We can use this to gate releases to force acknowledgment of releases. 
* Reuse of Azure Release to release from other branches.

Reply:
The target pipeline makes use of [Environments](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/environments?view=azure-devops) which provide the history,notification and approvals,
Any chance this addresses the listed concerns?

Reply:
I don't see artifacts being captured but that would be simple to add. Azure Releases and Azure Environments appear to share a lot of functionality. I have a question out to [public](https://stackoverflow.com/questions/76368866/azure-pipelines-releases-versus-environment-for-release-management-and-deployme) and internal SO on this. I think this is resolved for now.

Reply:
While Environments don't have all of the same functionality, it looks like that is the direction we should go. The answer posted on StackOverflow is very helpful.